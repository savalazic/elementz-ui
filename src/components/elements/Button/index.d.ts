export { ButtonProps, LoadingButtonProps } from './Button';
