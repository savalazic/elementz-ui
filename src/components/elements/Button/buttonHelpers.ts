export const getButtonColor = ({ type, theme }: { type: any; theme: any }) =>
  type ? theme[type] : theme.default;
