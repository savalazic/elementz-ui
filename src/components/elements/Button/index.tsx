export { Button, LoadingButton } from './Button';
