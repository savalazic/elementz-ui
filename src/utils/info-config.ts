// defaults for all stories
export const infoConfig = {
  info: {
    inline: true,
    header: false,
    styles: {
      infoBody: {
        marginTop: '50px',
        padding: '20px',
        lineHeight: '2',
      },
      source: {
        h1: {
          margin: '0',
        },
      },
    },
  },
};
