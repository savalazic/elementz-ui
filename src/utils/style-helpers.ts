export const truncateStyles = (width: string) => {
  return `
    width: ${width};
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  `;
};

export const setDimensions = (width: string, height: string) => {
  return `
    width: ${width};
    height: ${height};
  `;
};
