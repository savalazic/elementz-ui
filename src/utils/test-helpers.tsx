import React from 'react';
import { render } from 'react-testing-library';

import {
  ThemeProvider,
  defaultTheme,
} from '../components/utilities/ThemeProvider/ThemeProvider';

export function renderWithTheme(component: any) {
  return render(
    <ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>,
  );
}
