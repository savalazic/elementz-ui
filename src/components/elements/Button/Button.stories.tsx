import React from 'react';

import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs/react';
import { action } from '@storybook/addon-actions';

import { infoConfig } from '../../../utils/info-config';

import { Button, ButtonSizeEnum, ButtonTypeEnum } from './Button';

const stories = storiesOf('Elements|Button', module);

stories.addParameters(infoConfig);

stories
  .add('default', () => {
    const txt = text('Text', 'Button');
    return <Button onClick={action('onClick')}>{txt}</Button>;
  })
  .add('large', () => {
    const txt = text('Text', 'Button');
    return (
      <Button size={ButtonSizeEnum.large} onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('small', () => {
    const txt = text('Text', 'Button');
    return (
      <Button size={ButtonSizeEnum.small} onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('as anchor', () => {
    const txt = text('Text', 'Button');
    return (
      <Button href="https://github.com/savalazic/elementz-ui" target="_blank">
        {txt}
      </Button>
    );
  })
  .add('primary', () => {
    const txt = text('Text', 'Button');
    return (
      <Button type={ButtonTypeEnum.primary} onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('success', () => {
    const txt = text('Text', 'Button');
    return (
      <Button type={ButtonTypeEnum.success} onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('warning', () => {
    const txt = text('Text', 'Button');
    return (
      <Button type={ButtonTypeEnum.warning} onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('danger', () => {
    const txt = text('Text', 'Button');
    return (
      <Button type={ButtonTypeEnum.danger} onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('ghost primary', () => {
    const txt = text('Text', 'Button');
    return (
      <Button ghost type={ButtonTypeEnum.primary} onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('disabled', () => {
    const txt = text('Text', 'Button');
    return (
      <Button
        type={ButtonTypeEnum.primary}
        onClick={action('onClick')}
        disabled
      >
        {txt}
      </Button>
    );
  })
  .add('rounded', () => {
    const txt = text('Text', 'Button');
    return (
      <Button rounded onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('circle', () => {
    const size = text('Size', '50px');
    return (
      <Button circle={size} onClick={action('onClick')}>
        <span>üëåüèª</span>
      </Button>
    );
  })
  .add('fluid', () => {
    const txt = text('Text', 'Button');
    return (
      <Button fluid onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('text', () => {
    const txt = text('Text', 'Button');
    return (
      <Button text onClick={action('onClick')}>
        {txt}
      </Button>
    );
  })
  .add('loading', () => {
    const txt = text('Text', 'Button');
    const isLoading = boolean('Is loading', true);
    return (
      <Button
        type={ButtonTypeEnum.primary}
        loading={isLoading}
        onClick={action('onClick')}
      >
        {/* @TODO: implemenet loader, when loader component is ready */}
        <span>load</span>
        <span>{txt}</span>
      </Button>
    );
  });
