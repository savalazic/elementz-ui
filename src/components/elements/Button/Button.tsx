import React from 'react';
import styled from 'styled-components';
import { darken, lighten } from 'polished';

import { setDimensions, alignVertical } from '../../../utils/style-helpers';
import { getButtonColor } from './buttonHelpers';

export enum ButtonSizeEnum {
  normal = 'normal',
  small = 'small',
  large = 'large',
}

export enum ButtonTypeEnum {
  default = 'default',
  primary = 'primary',
  success = 'success',
  warning = 'warning',
  danger = 'danger',
}

export interface ButtonProps {
  /** @default normal */
  /** button size */
  size?: 'normal' | 'small' | 'large';
  /** button type */
  type?: 'default' | 'primary' | 'success' | 'warning' | 'danger';
  /** ghost */
  ghost?: boolean;
  /** rounded */
  rounded?: boolean;
  /** circle */
  circle?: string;
  /** fluid */
  fluid?: boolean;
  /** text */
  text?: boolean;
  /** loading */
  loading?: boolean;
  /** align vertical - used if button has more elements */
  alignVertical?: boolean;
  /** href link */
  href?: any;
  /** content */
  children: React.ReactNode;
}

const StyledButton = styled.button<ButtonProps>`
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  user-select: none;
  cursor: pointer;

  color: ${props =>
    (props.ghost && (props.type && props.theme[props.type])) || 'white'};
  color: ${props =>
    !props.ghost && props.type === ButtonTypeEnum.default && 'black'};
  color: ${props => props.text && props.theme.primary};

  background-color: ${props =>
    getButtonColor({ theme: props.theme, type: props.type })};
  background-color: ${props => props.ghost && 'transparent'};
  background-color: ${props => props.text && 'transparent'};

  opacity: ${props => (props.disabled || props.loading) && 0.7};

  border: ${props =>
    props.ghost &&
    `1px solid ${getButtonColor({ theme: props.theme, type: props.type })}`};

  border-radius: ${props => props.rounded && '20px'};
  border-radius: ${props => props.circle && '50%'};

  width: ${props => props.fluid && '100%'};

  ${props => props.alignVertical && alignVertical()}

  ${props => props.circle && setDimensions(props.circle, props.circle)}
  padding: ${props => props.circle && '0px'};
  line-height: ${props => props.circle && '100%'};

  font-size: ${props => {
    if (props.size === ButtonSizeEnum.normal) {
      return '14px';
    }
    if (props.size === ButtonSizeEnum.small) {
      return '10px';
    }
    if (props.size === ButtonSizeEnum.large) {
      return '20px';
    }
    return;
  }};

  pointer-events: ${props => (props.disabled || props.loading) && 'none'};

  transition: background-color 200ms, color 200ms;

  &:hover {
    background-color: ${props =>
      !props.ghost
        ? darken(0.1, getButtonColor({ theme: props.theme, type: props.type }))
        : getButtonColor({ theme: props.theme, type: props.type })};
    background-color: ${props => props.text && 'transparent'};
    color: ${props => props.ghost && 'white'};
    color: ${props => props.text && darken(0.1, props.theme.primary)}
  }

  &:focus {
    background-color: ${props =>
      lighten(0.05, getButtonColor({ theme: props.theme, type: props.type }))};
    background-color: ${props => props.text && 'transparent'};
    color: ${props => props.ghost && 'white'};
    color: ${props => props.text && lighten(0.05, props.theme.primary)}
  }

  &:active {
    background-color: ${props =>
      darken(0.1, getButtonColor({ theme: props.theme, type: props.type }))};
    background-color: ${props => props.text && 'transparent'};
    color: ${props => props.ghost && 'white'};
    color: ${props => props.text && lighten(0.05, props.theme.primary)}
  }
`;

export const Button = ({
  children,
  href,
  rounded,
  circle,
  fluid,
  ghost,
  type = 'default',
  disabled,
  text,
  loading,
  alignVertical: align,
  size = ButtonSizeEnum.normal,
  ...rest
}: ButtonProps &
  React.ButtonHTMLAttributes<HTMLButtonElement> &
  React.AnchorHTMLAttributes<HTMLAnchorElement>) => {
  return (
    <StyledButton
      as={href && 'a'}
      href={href}
      size={size}
      type={type}
      rounded={rounded}
      circle={circle}
      ghost={ghost}
      fluid={fluid}
      disabled={disabled}
      text={text}
      loading={loading}
      alignVertical={align}
      {...rest}
    >
      {children}
    </StyledButton>
  );
};

export default Button;
