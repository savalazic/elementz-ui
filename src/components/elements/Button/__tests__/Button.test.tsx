import 'jest-dom/extend-expect';
import React from 'react';
import { fireEvent } from 'react-testing-library';

import { Button } from '../Button';
import { renderWithTheme } from '../../../../utils/test-helpers';

describe('Button component', () => {
  const txt = 'Button';

  it('should render button without props', () => {
    const { container } = renderWithTheme(
      <Button type="primary">{txt}</Button>,
    );
    expect(container).toHaveTextContent(txt);
  });

  it('should call handleClick', () => {
    const handleClick = jest.fn();

    const { getByText } = renderWithTheme(
      <Button onClick={handleClick}>{txt}</Button>,
    );
    fireEvent.click(getByText(txt));
    fireEvent.click(getByText(txt));
    expect(handleClick).toHaveBeenCalledTimes(2);
  });

  it('should match default snapshot', () => {
    const { container } = renderWithTheme(<Button>{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `large` size prop', () => {
    const { container } = renderWithTheme(<Button size="large">{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `small` size prop', () => {
    const { container } = renderWithTheme(<Button size="small">{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `href` prop', () => {
    const { container } = renderWithTheme(
      <Button href="https://github.com/savalazic/elementz-ui">{txt}</Button>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `primary` type prop', () => {
    const { container } = renderWithTheme(
      <Button type="primary">{txt}</Button>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `success` type prop', () => {
    const { container } = renderWithTheme(
      <Button type="success">{txt}</Button>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `warning` type prop', () => {
    const { container } = renderWithTheme(
      <Button type="warning">{txt}</Button>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `danger` type prop', () => {
    const { container } = renderWithTheme(<Button type="danger">{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `ghost` prop', () => {
    const { container } = renderWithTheme(<Button ghost>{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `disabled` prop', () => {
    const { container } = renderWithTheme(<Button disabled>{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `rounded` prop', () => {
    const { container } = renderWithTheme(<Button rounded>{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `circle` prop', () => {
    const { container } = renderWithTheme(<Button circle="50px">{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `fluid` prop', () => {
    const { container } = renderWithTheme(<Button fluid>{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `text` prop', () => {
    const { container } = renderWithTheme(<Button text>{txt}</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with `loading` prop', () => {
    const { container } = renderWithTheme(<Button loading>{txt}</Button>);
    expect(container).toMatchSnapshot();
  });
});
