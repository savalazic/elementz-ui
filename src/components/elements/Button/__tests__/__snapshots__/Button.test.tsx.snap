// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match default snapshot 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`alignVerical\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    type="default"
  >
    <span>
      Test
    </span>
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`Button component should match snapshot with \`circle\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  padding: 0px;
  line-height: 100%;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`danger\` type prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  background-color: #f56c6c;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #f23c3c;
}

.c0:focus {
  background-color: #f78484;
}

.c0:active {
  background-color: #f23c3c;
}

<div>
  <button
    class="c0"
    type="danger"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`disabled\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  opacity: 0.7;
  font-size: 14px;
  pointer-events: none;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    disabled=""
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`fluid\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  width: 100%;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`ghost\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: #e1e1e1;
  background-color: #e1e1e1;
  background-color: transparent;
  border: 1px solid #e1e1e1;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #e1e1e1;
  color: white;
}

.c0:focus {
  background-color: #eee;
  color: white;
}

.c0:active {
  background-color: #c8c8c8;
  color: white;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`href\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <a
    class="c0"
    href="https://github.com/savalazic/elementz-ui"
    type="default"
  >
    Button
  </a>
</div>
`;

exports[`Button component should match snapshot with \`large\` size prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  font-size: 20px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`loading\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  opacity: 0.7;
  font-size: 14px;
  pointer-events: none;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`primary\` type prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  background-color: #409eff;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #0d84ff;
}

.c0:focus {
  background-color: #5aabff;
}

.c0:active {
  background-color: #0d84ff;
}

<div>
  <button
    class="c0"
    type="primary"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`rounded\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  border-radius: 20px;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`small\` size prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  background-color: #e1e1e1;
  font-size: 10px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
}

.c0:focus {
  background-color: #eee;
}

.c0:active {
  background-color: #c8c8c8;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`success\` type prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  background-color: #67c23a;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #529b2e;
}

.c0:focus {
  background-color: #76c94c;
}

.c0:active {
  background-color: #529b2e;
}

<div>
  <button
    class="c0"
    type="success"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`text\` prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  color: black;
  color: #409eff;
  background-color: #e1e1e1;
  background-color: transparent;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #c8c8c8;
  background-color: transparent;
  color: #0d84ff;
}

.c0:focus {
  background-color: #eee;
  background-color: transparent;
  color: #5aabff;
}

.c0:active {
  background-color: #c8c8c8;
  background-color: transparent;
  color: #5aabff;
}

<div>
  <button
    class="c0"
    type="default"
  >
    Button
  </button>
</div>
`;

exports[`Button component should match snapshot with \`warning\` type prop 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 8px 15px;
  width: auto;
  overflow: visible;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: inherit;
  -webkit-appearance: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: white;
  background-color: #e6a23c;
  font-size: 14px;
  -webkit-transition: background-color 200ms,color 200ms;
  transition: background-color 200ms,color 200ms;
}

.c0:hover {
  background-color: #d48a1b;
}

.c0:focus {
  background-color: #e9ad53;
}

.c0:active {
  background-color: #d48a1b;
}

<div>
  <button
    class="c0"
    type="warning"
  >
    Button
  </button>
</div>
`;
